### PR TITLE
Bug fixes

### LINKED ISSUE #1: Fixed value 183891
Hi, it is not clear for me why in the getVecs function,  
`if(vecs_type == 'Test'):
    index = index + 183891`
Is there any reason to put that value?

### PR DESCRIPTION
**Pull request by avinashsai**
#1 Use Infer_vectors:
Doc2Vec model should only be trained using training data and infer_vectors should be used for test
data.Hence modified the model to use only training data.
#2 random.shuffle()
This function shuffles the data given to doc2vec model in every epoch and hence the sentiment labels
will not map to their corresponding vectors. Hence modified get_vectors() function to map labels with
their corresponding vectors.

### CODE REVIEW

**Comment by avinashsai** [comment]
@ibrahimsharaf  please review this PR.

**Comment by avinashsai** [comment]
Hi @ibrahimsharaf  any progress on this??

**Comment by avinashsai** [comment]
Hi @ibrahimsharaf  what about this PR?

**Comment by avinashsai** [comment]
Hi @ibrahimsharaf status of this PR??

**Comment by ramaroberto** [comment]
Hi @avinashsai and @ibrahimsharaf, I think this PR might not be necessary:

#1 It's just a way to save time, if you check the example of gensim's doc2vec (https://github.com/RaRe-Technologies/gensim/blob/875b028ccd009fe2fa7665e177b8ab0b5e2dc40d/docs/notebooks/doc2vec-IMDB.ipynb), they're doing the same. It won't change much the final result, in fact, the result is worse than the experiment in the paper (I think this a gensim's problem), and if you compare your implementation to @ibrahimsharaf, the result doesn't change.

#2 It's shuffling all_data, which is a concatenation of x_train and x_test, hence a copy of the data (python does the copy when you concatenate the arrays). The x_train will remain unchanged and mapped to the labels besides all_data being shuffled. Also, the doc2vec model is building a map using the ids defined in  train_doc2vec for each document, so it doesn't matter if shuffle is applied since it will use that ID to restore the vector of the document from the map and continue the training. This map is also being used to restore the vectors on get_vectors function, so the ordering won't matter.

Also @ibrahimsharaf, I think that loop with the shuffle and alpha decreasing might not be necessary. I'm quite sure Doc2Vec.train is already shuffling and decreasing the alpha inside the method. What you are doing in the code is equivalent run Doc2Vec.train with 50 epochs.

Best,
Roberto.

**Comment by ibrahimsharaf** [comment]
Sorry @avinashsai for the long period, I think @ramaroberto comments make much sense (thanks!), closing this PR.
